### DISCUSSION TITLE
I have manged to get termux on wear os but....

### DISCUSSION THREAD

**Discussion by Asory2010** [ggml-org/llama.cpp#1378] [General]
I have manged to get termux on wear os but due to storage constrains i am looking for the smallest model supported plz help

**Comment by skidd-level-100** [comment]
This is funny! nice

If you are serous though there are some 70M parameter models on hugging face, around 100 mb

**Reply by Asory2010** [reply]
> This is funny! nice
> 
> If you are serous though there are some 70M parameter models on hugging face, around 100 mb

I am, but the 70m models don't work with llama.cpp or...?

**Comment by ghost** [comment]
70M parameters is extremely small. I'd be surprised if they did anything meaningfull, certainly they couldn't be used to run a general purpose AI assistant.

**Comment by BetaDoggo** [comment]
The smallest model supported would be llama 7b q4_0. Storage is a big issue though since if you barely have enough for the model you likely don't have any extra for disk offloading with swap, which will be required with how little ram most watches have.

**Reply by Asory2010** [reply]
I have 2gb of storage available and 4gb of ram (i am not sure about the ram). Is there any other projects that support linux/termux that support smaller models in chat mode?

**Reply by BetaDoggo** [reply]
I believe [koboldcpp](https://github.com/LostRuins/koboldcpp) supports some gpt2 models. There may be some tiny ones that could fit.

**Comment by Asory2010** [comment]
If anyone is interested in recreating this:
1. Sideload termux onto your watch
2.  Sideload unexpected keyboard
3.  Sideload The following app onto your watch (Keyboard_Del_Enter)
4. Now open termux and BOOM linux on your watch now all you have to do is apt update && apt upgrade and pkg install git and then clone the repo and git clone the model onto your watch and run it
